src/speaker.c

```diff
@@ -0,0 +1,68 @@
+#include "speaker.h"
+
+uint8_t BASE_NOTE = 9;
+uint8_t BASE_OCTAVE = 4;
+float BASE_FREQUENCY = 440.;
+uint8_t POSITIONS[8] = {9, 11, 0, 2, 4, 5, 7};
+
+/**
+ * @brief Set the frequency for the speaker.
+ *
+ * Frequency is set modulating the PWM signal sent to the speaker.
+ *
+ * @param[in] hz Frequency, in Hertz.
+ */
+static void speaker_set_frequency(float hz)
+{
+	uint16_t period;
+
+	period = (uint16_t)(SPEAKER_BASE_FREQUENCY_HZ / hz);
+	timer_set_period(TIM1, period);
+	timer_set_oc_value(TIM1, TIM_OC3, period / 2);
+}
+
+/**
+ * @brief Turn on the speaker to play the set frequency.
+ */
+static void speaker_on(void)
+{
+	timer_enable_counter(TIM1);
+	timer_enable_oc_output(TIM1, TIM_OC3);
+}
+
+/**
+ * @brief Turn off the speaker.
+ */
+static void speaker_off(void)
+{
+	timer_disable_counter(TIM1);
+	timer_disable_oc_output(TIM1, TIM_OC3);
+}
+
+/**
+ * @brief Play a note through the speaker.
+ *
+ * Even if this function uses the scientific notation, notes are played with
+ * the concert pitch (standard pitch). That means A above middle C is the
+ * reference note, played at 440 Hz.
+ *
+ * @param[in] note Which note to play, in scientific notation.
+ * @param[in] octave Which octave to play, in scientific notation.
+ * @param[in] accidental Number of semitones to sum to the note.
+ * @param[in] duration Duration of the note, in seconds.
+ */
```

Comment:
Include a @note, explaining that the the speaker and the emitters are sharing the timer 1 so they can not work simultaneously.

Reply:
Fixed in 6383d53035b08cf26c979dcda0ab48d79f3d0303